### PR TITLE
[state sync] specs/README

### PR DESCRIPTION
**Pull request by christinacelee**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/libra/developers.libra.org, and link to your PR here.)

### CODE REVIEW

File: specifications/state_sync/README.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+as a node in the Libra Network
+4. Abstracted Modules: external components that state synchronizer interfaces with
+
+
+## Overview
+
+State Synchronizer is a system component that is responsible for syncing ledger
+state among peers in the Libra Payment Network. Using state sync, a Libra node
+can discover and update to a more recent ledger state if available with the following
+general flow:
```
Comment (davidiw) [inline specifications/state_sync/README.md:23]: while this is technically correct, it is probably worth stating that it synchronizes the ledger by sharing all transactions and proof that the transactions have been processed by consensus or something. Then state sync re-execute the transactions to actually synchronize the ledger.

File: specifications/state_sync/README.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
+# State Synchronizer specification
+
+## Abstract
+
+This document describes the spec of a Libra node's state synchronizer component.
```
Comment (davidiw) [inline specifications/state_sync/README.md:5]: could we give a one liner here ... ", which is responsible for synchronizing the local ledger to the latest state in the LPN or whatever".

then state what this implementation does.

**Review by davidiw — APPROVED** [review]
The description of the protocol seems like it should be simplified... I probably need to look at the code more carefully but do validator and fn state sync really differ /that/ much that there's really little overlap?

File: state-synchronizer/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+The State Synchronizer is the component that helps nodes advance its local 
+ledger state by requesting chunks of transactions from peers that have more 
+up-to-date state. A node sends a chunk request to its peer, and the peers 
+responds with data that helps the node advance its local state. 
+
+## Overview
+
+Refer to the state synchronizer specifications for an high-level overview:
+https://github.com/libra/libra/blob/master/specifications/state_sync/spec.md
```
[the remark this answers is not in the document]
Comment (mimoo) [reply state-synchronizer/README.md:16]: better:

```
Refer to the [state synchronizer specification](../specifications/state_sync) for an high-level overview.
```

File: state-synchronizer/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+and sending chunk requests to advance local state based on per-peer info or network 
+health
+- `synchronizer`: interface that manages state sync's interactions with consensus
+
+
+## How is this module organized?
+```
+state-synchronizer
+|- src
+   |-- tests/                          # Unit/integration tests
```
Comment (mimoo) [inline state-synchronizer/README.md:38]: bad spacing?

File: specifications/state_sync/README.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+general flow:
+1. A node sends a [`GetChunkRequest`](#getchunkrequest) to a remote peer
+2. If there is a more recent ledger state available on the remote
+peer, it sends back a corresponding [`GetChunkResponse`](#getchunkresponse) with data that helps
+the requester advance its local ledger state
+
+
+There are three types of supported sync protocols:
+1. [Validator sync](#validator-sync): In general, validators learn about recent
+ledger state through their participation in the consensus protocol(TODO link),
```
Comment (mimoo) [inline specifications/state_sync/README.md:32]: [consensus protocol](../specifications/consensus)

File: specifications/state_sync/README.md [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+network partition). In these cases, a lagging validator can rely on state
+synchronization to catch up to the more recent state. 
+
+2. [Full Node sync](#full-node-sync): As full nodes do not participate in consensus,
+they rely only on the state synchronizer to discover and sync to a more recent
+ledger state. 
+
+3. [Waypoint sync](#waypoint-sync): When a node starts up, it can specify a waypoint
+to sync its local state to - see section 3.4 in [this paper](https://github.com/libra/libra/blob/master/documentation/tech-papers/lbft-verification/lbft-verification.pdf) for more context and 
+motivation for waypoint sync, as this specification primariliy focuses on
```
Comment (mimoo) [inline specifications/state_sync/README.md:43]: "primariliy"

File: specifications/state_sync/README.md [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+the technical requirements for the waypoint sync protocol. Waypoint sync 
+happens at node initialization and must be completed before the node participates
+in other state sync protocols. 
+
+A properly implemented State Synchronizer component supports each
+sync protocol as both the requester and the responder. See [Sync Protocols](#sync-protocols)
+for the technical requirements for each of these sync protocols.
```
Comment (mimoo) [inline specifications/state_sync/README.md:50]: perhaps a note on the fact that a single state-sync runs for all of these 3 points when you're a validator (as opposed to one state-sync for validator network and one state-sync for FN network)

**Pull request closed without merging**